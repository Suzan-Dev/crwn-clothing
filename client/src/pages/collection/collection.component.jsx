import React from 'react';
import { connect } from 'react-redux';

import {
  CollectionPageContainer,
  TitleContainer,
  ItemsContainer,
  CollectionItemContainer,
} from './collection.style';

import { selectCollections } from '../../redux/shop/shop.selector';

export const CollectionPage = ({ collection }) => {
  const { title, items } = collection;
  return (
    <CollectionPageContainer>
      <TitleContainer className='title'>{title}</TitleContainer>
      <ItemsContainer>
        {items.map((item) => (
          <CollectionItemContainer key={item.id} item={item} />
        ))}
      </ItemsContainer>
    </CollectionPageContainer>
  );
};

const mapStateToProps = (state, ownProps) => ({
  collection: selectCollections(ownProps.match.params.collectionId)(state),
});

export default connect(mapStateToProps)(CollectionPage);
