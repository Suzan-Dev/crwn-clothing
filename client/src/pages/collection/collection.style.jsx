import styled from 'styled-components';
import CollectionItem from '../../components/collection-item/collection-item.component';

export const CollectionPageContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export const TitleContainer = styled.div`
  font-size: 38px;
  margin: 0 auto 30px;
`;

export const ItemsContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 10px;

  @media screen and (min-width: 800px) {
    grid-template-columns: repeat(4, 1fr);
  }
`;

export const CollectionItemContainer = styled(CollectionItem)`
  margin-bottom: 30px;
`;
