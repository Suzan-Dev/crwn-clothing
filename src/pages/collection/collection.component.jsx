import React from 'react';
import { connect } from 'react-redux';

import './collection.style.scss';

import { selectCollections } from '../../redux/shop/shop.selector';
import CollectionItem from '../../components/collection-item/collection-item.component';

const CollectionPage = ({ collection }) => {
  const { title, items } = collection;
  return (
    <div className='collection-page'>
      <div className='title'>{title}</div>
      <div className='items'>
        {items.map((item) => (
          <CollectionItem key={item.id} item={item} />
        ))}
      </div>
    </div>
  );
};

const mapStateToProps = (state, ownProps) => ({
  collection: selectCollections(ownProps.match.params.collectionId)(state),
});

export default connect(mapStateToProps)(CollectionPage);
