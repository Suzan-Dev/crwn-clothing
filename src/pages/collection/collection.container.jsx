import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import WithSpinner from '../../components/with-spinner/with-spinner.component';
import CollectionPage from './collection.component';

import { selectIsCollectionLoaded } from '../../redux/shop/shop.selector';

const mapStateToProps = createStructuredSelector({
  isLoading: (state) => !selectIsCollectionLoaded(state),
});

const CollectionPageContainer = compose(
  connect(mapStateToProps),
  WithSpinner
)(CollectionPage);

export default CollectionPageContainer;
