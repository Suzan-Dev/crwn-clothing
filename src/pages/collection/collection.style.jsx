import styled from 'styled-components';
import CollectionItem from '../../components/collection-item/collection-item.component';

export const CollectionPageContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export const TitleContainer = styled.div`
  font-size: 38px;
  margin: 0 auto 30px;
`;

export const ItemsContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 10px;
`;

export const CollectionItemContainer = styled(CollectionItem)`
  margin-bottom: 30px;
`;
