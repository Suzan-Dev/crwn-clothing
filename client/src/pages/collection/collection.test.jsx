import React from 'react';
import { shallow } from '../../setupTests';
import { CollectionPage } from './collection.component';

describe('Testing: CollectionPage component!', () => {
  const mockCollection = {
    title: 'Test',
    items: [{ id: 1 }, { id: 2 }],
  };

  const mockProps = {
    collection: mockCollection,
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CollectionPage {...mockProps} />);
  });

  test('should render CollectionPage component.', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should render title prop correctly.', () => {
    expect(wrapper.find('.title').text()).toBe(mockCollection.title);
  });

  test('should render the same number of CollectionItems as in collection array.', () => {
    expect(wrapper.find('Styled(Connect(CollectionItem))').length).toBe(mockCollection.items.length);
  });
});
