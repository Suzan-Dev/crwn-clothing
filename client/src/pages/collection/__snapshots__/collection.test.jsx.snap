// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing: CollectionPage component! should render CollectionPage component. 1`] = `
"<styled.div>
  <styled.div className=\\"title\\">
    Test
  </styled.div>
  <styled.div>
    <Styled(Connect(CollectionItem)) item={{...}} />
    <Styled(Connect(CollectionItem)) item={{...}} />
  </styled.div>
</styled.div>"
`;
